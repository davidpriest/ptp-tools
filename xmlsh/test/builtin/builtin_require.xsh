# Test require command
# First implemented in version 1.0.1

echo entering require $*
# If called recursively then fail conditionally and explictily

if [ $# -gt 0 ] ; then 
	require 99999 || echo Correctly failed require 99999
	require 99999
	echo SNH - require should have exited 
	exit 1
fi

require	# should succeed
require || echo Require with no args should have succeeded
require && echo Require succeeded with no args 
require 1.0.1 
require 1.0.1 && echo Require 1.0.1 succeeded
require 9.0.2 || echo Require 9.0.2 failed correctly

# should fail - trap it
try {
	$0 test 
} catch  e 
{ 
	echo Correctly caught failed require in sub 
	echo $e 
}
