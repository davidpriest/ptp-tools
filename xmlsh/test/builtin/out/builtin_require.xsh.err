requires version: 99999
