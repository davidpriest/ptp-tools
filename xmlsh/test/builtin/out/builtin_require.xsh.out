entering require
Require succeeded with no args
Require 1.0.1 succeeded
Require 9.0.2 failed correctly
entering require test
Correctly failed require 99999
Correctly caught failed require in sub
requires version: 99999
